### PR TITLE
Handling of LogicalNotRequest implemented

### LINKED ISSUE #730: Revit_Toolkit: implement LogicalNotRequest
#### Description:

Following https://github.com/BHoM/BHoM/issues/889

### PR DESCRIPTION
**Pull request by pawelbaran**
### NOTE: Depends on 

https://github.com/BHoM/BHoM/pull/890
https://github.com/BHoM/BHoM_Engine/pull/1798

   
### Issues addressed by this PR

Closes #730

### Test files

On [SharePoint](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?RootFolder=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F01%5FTest%20Scripts%2FRevit%5FToolkit%2FRevit%5FToolkit%2DIssue730%2DLogicalNotRequest&FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4)

### Additional comments

There might be a minor conflict between this PR and #726 - happy to resolve it manually, first merge first serve.

### CODE REVIEW

**Comment by pawelbaran** [comment]
I had the same concern @kThorsager, but came to a conclusion that we cannot stop the user from shooting her/his own feet (as we say in Polish).

File: Engine_Revit_UI/Query/ElementIds/ElementIdsByExclusion.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        [Input("toExclude", "List of ElementIds to be excluded from the output.")]
+        [Input("ids", "Optional, allows narrowing the search: if not null, the output will be an intersection of this collection and ElementIds filtered by the query.")]
+        [Output("elementIds", "Collection of filtered ElementIds.")]
+        public static IEnumerable<ElementId> ElementIdsByExclusion(this Document document, IEnumerable<ElementId> toExclude, IEnumerable<ElementId> ids = null)
+        {
+            if (document == null)
+                return null;
+
+            if (ids != null && ids.Count() == 0)
+                return new List<ElementId>();
```
Comment (AndreBalasian) [inline Engine_Revit_UI/Query/ElementIds/ElementIdsByExclusion.cs:49]: Add a warning message regarding the return of an empty list

Comment (pawelbaran) [reply Engine_Revit_UI/Query/ElementIds/ElementIdsByExclusion.cs:49]: That is fine, there is nothing wrong in no elements passing a certain filter I believe

**Comment by pawelbaran** [comment]
Hmm, you should be getting the outputs before bypassing as well. That is weird - did it look the same for you @kThorsager?

**Comment by kThorsager** [comment]
I definitely remember getting outputs before changing thing up.

**Comment by pawelbaran** [comment]
Then it means there might be some build issues on your end @AndreBalasian - did you rebuild BHoM, BHoM_Engine and Revit_Toolkit on branches relevant to this PR?

**Comment by AndreBalasian** [comment]
Have just rebuilt all dependencies, including the GS toolkit, and the pre-bypass output looks better.
image

**Pull request merged**